contracts/test/_global-hooks.js

```diff
@@ -34,11 +39,25 @@ mocha.before(function () {
 
   // If you are running unit tests, scrape out all fork tests.
   // For fork tests, scrape out all unit tests.
-  root.suites = root.suites.filter(
-    (s) =>
-      s.file.endsWith(".fork-test.js") == isForkTest &&
-      s.file.endsWith(".arb.fork-test.js") == isArbFork
-  );
+  root.suites = root.suites.filter((s) => {
+    const isMainnetForkTestFile = s.file.endsWith("mainnet.fork-test.js");
+    const isHoleskyTestFile = s.file.endsWith(".holesky.fork-test.js");
+    const isArbTestFile = s.file.endsWith(".arb.fork-test.js");
+    const isBaseTestFile = s.file.endsWith(".base.fork-test.js");
+
+    if (isArbFork) {
+      return isArbTestFile;
+    } else if (isMainnetForkTest) {
+      return isMainnetForkTestFile && !isArbTestFile;
+    } else if (isBaseFork) {
+      return isBaseTestFile;
+    } else if (isHoleskyFork) {
+      return isHoleskyTestFile;
+    } else {
+      // else is unit test
+      return !isMainnetForkTestFile && !isHoleskyTestFile;
```

Comment:
does it require `!isArbTestFile && !isBaseTestFile`?

Reply:
great catch! corrected here: https://github.com/OriginProtocol/origin-dollar/pull/2094/commits/877cc11c84fe9019c0b2f6cfd3e50814177aa582

Reply:
much cleaner, thanks!